httpclient/logging_round_tripper.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright © 2024 Acronis International GmbH.
+
+Released under MIT license.
+*/
+
+package httpclient
+
+import (
+	"context"
+	"fmt"
+	"github.com/acronis/go-appkit/httpserver/middleware"
+	"github.com/acronis/go-appkit/log"
+	"net/http"
+	"time"
+)
+
+// LoggerMode represents a mode of logging.
+type LoggerMode string
+
+const (
+	LoggerModeNone   LoggerMode = "none"
+	LoggerModeAll    LoggerMode = "all"
+	LoggerModeFailed LoggerMode = "failed"
+)
+
+// IsValid checks if the logger mode is valid.
+func (lm LoggerMode) IsValid() bool {
+	switch lm {
+	case LoggerModeNone, LoggerModeAll, LoggerModeFailed:
+		return true
+	}
+	return false
+}
+
+// LoggingRoundTripper implements http.RoundTripper for logging requests.
+type LoggingRoundTripper struct {
+	// Delegate is the next RoundTripper in the chain.
+	Delegate http.RoundTripper
+
+	// ReqType is a type of request.
+	ReqType string
+
+	// Opts are the options for the logging round tripper.
+	Opts LoggingRoundTripperOpts
```

Comment:
In go-appkit optional structs are usually used only for initialization, but objects stores all dependencies as direct fields. Example: https://github.com/acronis/go-appkit/blob/master/httpclient/retryable_round_tripper.go#L45